### DISCUSSION TITLE
How to extend userAvatar instead of replacing?

### DISCUSSION THREAD

**Discussion by DanMossa** [flyerhq/flutter_chat_ui#329] [Q&A] [answered]
I'd like to put the userAvatar inside of a card so that it looks clickable.

I'm having trouble just adding a background?

**Comment by demchenkoalex** [comment]
Hi! Is there a way to extend widgets? Do you have an example maybe?

**Reply by DanMossa** [reply]
@demchenkoalex I think I'm imagining something like a `.copyFrom` method.

https://developer.school/tutorials/dart-flutter-what-does-copywith-do

**Reply by demchenkoalex** [reply, marked as answer]
But this is for the data classes, avatar is a widget, not a data class. How would you modify properties to add another widget on top of something with copyWith?
